mlir/lib/Dialect/Rock/Transforms/ViewToTransform.cpp

```diff
@@ -86,6 +86,50 @@ struct ExpandShapeRewritePattern
   }
 };
 
+struct ExtractSliceRewritePattern
+    : public OpConversionPattern<tensor::ExtractSliceOp> {
+  using OpConversionPattern<tensor::ExtractSliceOp>::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(tensor::ExtractSliceOp sliceOp,
+                                OpAdaptor adaptor,
+                                ConversionPatternRewriter &b) const final {
+    Location loc = sliceOp.getLoc();
+    ArrayRef<int64_t> inpShape = sliceOp.getSourceType().getShape();
+    ArrayRef<int64_t> outShape = sliceOp.getResult().getType().getShape();
+
+    for (size_t i = 0; i < sliceOp.getSizes().size(); i++) {
+      if (sliceOp.isDynamicOffset(i)) {
+        return sliceOp.emitError("dynamic offsets are not supported.");
+      }
+      if (sliceOp.isDynamicSize(i)) {
+        return sliceOp.emitError("dynamic sizes are not supported.");
+      }
+      if (sliceOp.isDynamicStride(i)) {
+        return sliceOp.emitError("dynamic sizes are not supported.");
+      }
+    }
+
+    ArrayRef<int64_t> offsets = sliceOp.getStaticOffsets();
+    ArrayRef<int64_t> sizes = sliceOp.getStaticSizes();
+    ArrayRef<int64_t> strides = sliceOp.getStaticStrides();
+
+    for (int64_t stride : strides) {
+      if (stride != 1) {
```

Comment:
Let's go ahead and support it? "pick every Nth element" is the function `(..., i, ...) -> (..., N * i, ...)`, which is just an Embed{}. This'd also let us support dilations in the Tosa interchange layer, I think.

The way I'd think to do this is to collect up all the non-unit slice dimensions, and then add a second `rock.transform` that applies the strides.

Reply:
agreed with the suggestion though I dont want to do that given the on going things.
I've created a ticket for next sprint so that we dont forget  : https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/870